ssa/utils/cert_validation.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"encoding/base64"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/transport"
+)
+
+// ValidateCABundle validates a CA bundle using the same logic as Kubernetes 1.31+.
+// Returns true if the CA bundle is valid, false otherwise.
+func ValidateCABundle(caBundle []byte) bool {
```

Comment:
This function is not used outside `ssa/utils` at the moment, I'd make it not exported and test only the exported functions that use it internally